src/Altinn.Notifications/Controllers/SmsNotificationOrdersController.cs

```diff
@@ -0,0 +1,108 @@
+/// </summary>
+[Route("notifications/api/v1/orders/sms")]
+[ApiController]
+[Authorize(Policy = AuthorizationConstants.POLICY_CREATE_SCOPE_OR_PLATFORM_ACCESS)]
+[SwaggerResponse(401, "Caller is unauthorized")]
+[SwaggerResponse(403, "Caller is not authorized to access the requested resource")]
+
+public class SmsNotificationOrdersController : ControllerBase
+{
+    private readonly IValidator<SmsNotificationOrderRequestExt> _validator;
+    private readonly IOrderRequestService _orderRequestService;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SmsNotificationOrdersController"/> class.
+    /// </summary>
+    public SmsNotificationOrdersController(IValidator<SmsNotificationOrderRequestExt> validator, IOrderRequestService orderRequestService)
+    {
+        _validator = validator;
+        _orderRequestService = orderRequestService;
+    }
+
+    /// <summary>
+    /// Add an SMS notification order.
+    /// </summary>
+    /// <remarks>
+    /// The API will accept the request after som basic validation of the request.
+    /// The system will also attempt to verify that it will be possible to fulfill the order.
+    /// </remarks>
+    /// <returns>The id of the registered notification order</returns>
+    [HttpPost]
+    [Consumes("application/json")]
+    [Produces("application/json")]
+    [SwaggerResponse(202, "The notification order was accepted", typeof(OrderIdExt))]
+    [SwaggerResponse(400, "The notification order is invalid", typeof(ValidationProblemDetails))]
+    [SwaggerResponseHeader(202, "Location", "string", "Link to access the newly created notification order.")]
+    public async Task<ActionResult<OrderIdExt>> Post(SmsNotificationOrderRequestExt smsNotificationOrderRequest)
+    {
+        FluentValidation.Results.ValidationResult validationResult = _validator.Validate(smsNotificationOrderRequest);
+        if (!validationResult.IsValid)
+        {
+            validationResult.AddToModelState(this.ModelState);
+            return ValidationProblem(ModelState);
+        }
+
+        string? creator = HttpContext.GetOrg();
+
+        if (creator == null)
+        {
+            return Forbid();
+        }
+
+        NotificationOrderRequest orderRequest = MapToOrderRequest(smsNotificationOrderRequest, creator);
+        (NotificationOrder? registeredOrder, ServiceError? error) = await _orderRequestService.RegisterNotificationOrder(orderRequest);
+
+        if (error != null)
+        {
+            return StatusCode(error.ErrorCode, error.ErrorMessage);
+        }
+
+        string selfLink = registeredOrder!.GetSelfLink();
+        return Accepted(selfLink, new OrderIdExt(registeredOrder!.Id));
+    }
+
+    /// <summary>
+    /// Maps a <see cref="SmsNotificationOrderRequestExt"/> to a <see cref="NotificationOrderRequest"/>
+    /// </summary>
+    public static NotificationOrderRequest MapToOrderRequest(SmsNotificationOrderRequestExt extRequest, string creator)
```

Comment:
I would like to see this logic in a separate mapper class or as a part of one of the classes we're mapping between. Makes more sense to call `NotificationOrderRequest` or `SmsNotificationOrderRequestExt` than `SmsNotificationsOrdersControllers` when needing to map between the two models outside of the controller.

Reply:
My actual intention is to make this `MapToOrderRequest` a `private` non-static method of the same class. Why? because it's curated specifically for this class. And it isn't used anywhere else. So why should we overload this method unnecessary! 🤔 The same goes for the `MapToOrderRequest` overloaded method for `EmailNotificationOrderRequestExt` which we are using in a class and it's more suitable for being a helper method than overloaded one. 🤔 So for the sake of maintainability wouldn't it be better to use them as helper method? 🤔  

https://github.com/Altinn/altinn-notifications/blob/cfa975c301bc8003eeb8058e008bb8b7303ff68e/src/Altinn.Notifications/Mappers/OrderMapper.cs#L19

Reply:
Making the method private will not allow us to test the mapping isolated, which I think would be a good thing if we could. Grouping mapping logic together in a class will allow us to only navigate to a single file to find all mappings related to the order request. My point for view is that we can benefit by sharing the logic for mapping between the more complex shared types in the order request e.g. recipient. We will have more complex orders in the future that will require addressing both sms and email properties in the same mapping.